test/OpenTelemetry.Tests/Metrics/AggregatorTest.cs

```diff
@@ -104,6 +104,34 @@ public void HistogramDistributeToAllBucketsCustom()
             Assert.Equal(boundaries.Length + 1, actualCount);
         }
 
+        [Fact]
+        public void HistogramBinaryBucketTest()
+        {
+            // Arrange
+            var boundaries = new double[HistogramBuckets.DefaultHistogramCountForBinarySearch];
+            for (var i = 0; i < boundaries.Length; i++)
+            {
+                boundaries[i] = i;
+            }
+
+            var histogramPoint = new MetricPoint(this.aggregatorStore, AggregationType.Histogram, null, null, boundaries);
+
+            // Act
+            histogramPoint.Update(-1);
```

Comment:
Check for edge cases too: `double.NegativeInfinity`, `boundaries[0]`, `boundaries[boundaries.Length -1]`, `double.PositiveInfinity`.

Reply:
I believe that infinity is not a value that the spec defines which bucket it belongs in. To make sure -Inf goes in the first bucket and +Inf goes in the last bucket it would require additional checks and slowdown the program. I will add the other edge ones you mention, but I think the infinities need to be revisited at a later time. I don't want to write a test that includes them as making that test pass would put both in the final bucket with the current implementation.